### PR TITLE
Add a bunch of fields to RtpParameters and RtpEncodingParameters

### PR DESCRIPTION
**Pull request by pthatcherg**
This has had little to no list discussion so far.  This is intended to start the discussion :).

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -2775,8 +2775,39 @@
...
+              May not be changed by the application.
+            </p>
+          </dd>
+
+          <dt>RTCRtxParameters rtx</dt>
+          <dd>
+            <p>The parameters used for RTX, or unset if RTX is not being used.</p>
+          </dd>
+
+          <dt>RTCRtxParameters fec</dt>
```
Comment (juberti) [inline webrtc.html]: RTCFecParameters?

File: webrtc.html [hunk 2]
```diff
@@ -2794,6 +2825,99 @@
...
+              The value put in the RTP packet to identify the header
+              extension.  Effectively readonly.  May not be changed by
+              the application.
+            </p>
+          </dd>
+
+          <dt>boolean encrypt</dt>
+          <dd>
+            <p>
+              Whether the header extension is encryted or not.
```
Comment (juberti) [inline webrtc.html]: encrypted

File: webrtc.html (hunk 1 above)
Comment (pthatcherg) [inline webrtc.html]: Yep, fixed.

File: webrtc.html (hunk 2 above)
Comment (pthatcherg) [inline webrtc.html]: I changed it to "encrypted", but we should be awkward later when we want to make it writable (as it is in ORTC, where it's named "encrypt".

File: webrtc.html [hunk 3]
```diff
@@ -2775,8 +2775,39 @@
...
+          </dd>
         </dl>
+
         <dl class="idl" title="dictionary RTCRtpEncodingParameters">
+          <dt>unsigned long ssrc</dt>
+          <dd>
+            <p>
+              The SSRC of the RTP source stream of this encoding
+              (non-RTX, non-FEC RTP stream).  Effectively readonly.
+              May not be changed by the application.
```
Comment (ekr) [inline webrtc.html]: I suggest just saying "readonly"  and then explaining in the text what happens if you try to change it.

Comment (pthatcherg) [inline webrtc.html]: Done

File: webrtc.html [hunk 4]
```diff
@@ -2775,8 +2775,40 @@
           <dd>
             <p>An array containing parameters for RTP encodings of media.</p>
           </dd>
+
+          <dt>sequence&lt;RTCRtpHeaderExtensionParameters&gt; headerExtensions</dt>
+          <dd>
+            <p>An array containing parameters for RTP header extensions.</p>
```
Comment (jan-ivar) [inline webrtc.html]: s/array/sequence/

File: webrtc.html [hunk 5]
```diff
@@ -2794,6 +2826,106 @@
              in [[!RTCWEB-TRANSPORT]], Section 4.
             </p>
           </dd>
+
+          <dt>RTCDegregationPreference degregationPreference = "balanced"</dt>
```
Comment (jan-ivar) [inline webrtc.html]: Degradation

File: webrtc.html [hunk 6]
```diff
@@ -2794,6 +2826,106 @@
...
+          </dd>
+        </dl>
+
+        <dl class="idl" title="dictionary RTCRtxParameters">
+          <dt>unsigned long ssrc</dt>
+          <dd>
+            <p>
+              The SSRC of the RTP stream for RTX.  Read-only.
+              Read-only.  If <codec>RTCRtpSender.setParameters</codec>
+              is called with an altered value, an exception will be
```
Comment (jan-ivar) [inline webrtc.html]: Which exception? Did we want to do s/dictionary/interface/ here?

File: webrtc.html [hunk 7]
```diff
@@ -2775,8 +2775,40 @@
           <dd>
             <p>An array containing parameters for RTP encodings of media.</p>
           </dd>
+
+          <dt>sequence&lt;RTCRtpHeaderExtensionParameters&gt; headerExtensions</dt>
+          <dd>
+            <p>An array containing parameters for RTP header extensions.</p>
+          </dd>
+
+          <dt>RTCRtcpParameters</dt>
```
Comment (dontcallmedom) [inline webrtc.html]: misses the name of the dictionary member (`parameters`?)

**Comment by alvestrand** [comment]
Met with approval in Seattle. Should be merged.

File: webrtc.html (hunk 4 above)
Comment (pthatcherg) [inline webrtc.html]: Done

File: webrtc.html (hunk 5 above)
Comment (pthatcherg) [inline webrtc.html]: done

File: webrtc.html (hunk 7 above)
Comment (pthatcherg) [inline webrtc.html]: Done (it's "rtcp").

File: webrtc.html (hunk 6 above)
Comment (pthatcherg) [inline webrtc.html]: How about InvalidParameter?

**Comment by adam-be** [comment]
@pthatcherg, can you rebase this one please

**Comment by pthatcherg** [comment]
Merged (I hope that's as good as rebased)

**Comment by adam-be** [comment]
Perfect, thanks

**Comment by adam-be** [comment]
It looks like maxBitrate has moved from RTCRtpEncodingParameters to RTCRtpHeaderExtensionParameters. But I'll fix that before I merge it.

**Comment by adam-be** [comment]
@pthatcherg, I made a PR against your PR with two proposals. The first fixes the merge issue I mentioned above. The second centralizes error handling where it simply marks each read-only param and describes what to do in the setParameters() function. Also I couldn't find any "InvalidParameter" in the WebIDL spec [1] so I switched to a "ReadOnlyError" instead.

[1] http://heycam.github.io/webidl/#idl-DOMException-error-names

**Comment by pthatcherg** [comment]
Thanks, Adam, for fixing my merge mistake.  I merged your change.  Hopefully this PR is in good shape now.

**Comment by adam-be** [comment]
I merged with the "ReadOnlyError" in place to get this included in the next release. It's only mentioned at one location now so it's easy to change if we decide to use some other type.

**Pull request merged**